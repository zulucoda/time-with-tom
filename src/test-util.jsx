import React from 'react';
import { oneOfType, shape, element, func } from 'prop-types';
import { Form, Formik } from 'formik';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * Formik Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {Object} props
 * @param {Object} initialValues
 * @param {Function} onSubmit
 * @returns {Formik}
 */
export const FormikWrapperTestUtil = ({
  Component,
  props = {},
  initialValues = {},
  onSubmit = jest.fn(),
}) => (
  <Formik initialValues={initialValues} onSubmit={onSubmit}>
    <Form>
      <Component {...props} />
    </Form>
  </Formik>
);

FormikWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
  initialValues: shape({}),
  onSubmit: func,
};

/**
 * React Router Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {object} props
 * @returns {React.FunctionComponent}
 */
export const ReactRouterWrapperTestUtil = ({ Component, props = {} }) => (
  <Router>
    <Component {...props} />
  </Router>
);

ReactRouterWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
};
