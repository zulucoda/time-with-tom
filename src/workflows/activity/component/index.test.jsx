import React from 'react';
import { render } from '@testing-library/react';
import Activity from './index';
import { FormikWrapperTestUtil } from '../../../test-util';

describe('Activity component', function () {
  test('does it render', () => {
    const { container } = render(
      <FormikWrapperTestUtil Component={Activity} />,
    );
    expect(container).toMatchSnapshot();
  });
});
