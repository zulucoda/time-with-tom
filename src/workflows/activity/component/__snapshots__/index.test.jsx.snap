// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity component does it render 1`] = `
<div>
  <form
    action="#"
  >
    <div>
      <div
        class="sc-AxjAm hcKRzz"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        >
          activity
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-activity"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="mui-component-select-activity"
            role="button"
            tabindex="0"
          >
            <span>
              â€‹
            </span>
          </div>
          <input
            name="activity"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </form>
</div>
`;
